In [ ]:
%run training.ipynb

In [6]:
#Now evaluating our models accuracy 
#Importing necessary libraries
from sklearn.metrics import accuracy_score,f1_score,classification_report
from sklearn.metrics import confusion_matrix

In [7]:
#Accuracy score for logistic  regression 
acc_score_lr=accuracy_score(y_test,y_pred)
print('Accuracy score for logistic regression is:',acc_score_lr)
f1_Score_lr=f1_score(y_test,y_pred)
print('F1 score for logistic regression is:',f1_Score_lr)

Accuracy score for logistic regression is: 0.9527511961722488
F1 score for logistic regression is: 0.784741144414169


In [8]:
#Accuracy score for Naive  Bayes 
acc_score_nb=accuracy_score(y_test,y1_pred)
print('Accuracy score for Naive Bayes is:',acc_score_nb)
f1_Score_nb=f1_score(y_test,y1_pred)
print('F1 score for Naive Bayes is:',f1_Score_nb)

Accuracy score for Naive Bayes is: 0.9712918660287081
F1 score for Naive Bayes is: 0.8775510204081632


In [9]:
#Now building confusion matrix
cm1=confusion_matrix(y_test,y_pred)
print("The confusion matrix for logistic regression is:\n",cm1)
cm2=confusion_matrix(y_test,y1_pred)
print("The confusion matrix for Naive Bayes is:\n",cm2)

The confusion matrix for logistic regression is:
 [[1449    4]
 [  75  144]]
The confusion matrix for Naive Bayes is:
 [[1452    1]
 [  47  172]]


In this project, SMS spam classification was performed using machine learning techniques after applying text preprocessing and feature extraction methods. The cleaned text data was transformed into numerical form and used to train Logistic Regression and Naive Bayes models. The experimental results showed that Naive Bayes significantly outperformed Logistic Regression, achieving an accuracy of 97.13% and an F1 score of 0.88, compared to 95.27% accuracy and 0.78 F1 score for Logistic Regression. This indicates that Naive Bayes is more effective in handling text data and class imbalance, making it more suitable for spam detection tasks. Further improvements can be made by tuning feature extraction techniques and experimenting with more advanced models.
